packages/contracts/contracts/voting/majority/MajorityVotingBase.sol

```diff
@@ -54,8 +53,74 @@ abstract contract MajorityVotingBase is
     Initializable,
     ERC165Upgradeable,
     TimeHelpers,
-    PluginUUPSUpgradeable
+    GovernancePluginUUPSUpgradeable
 {
+    /// @notice The different voting modes available.
+    /// @param Standard In standard mode, early execution and vote replacement are disabled.
+    /// @param EarlyExecution In early execution mode, a proposal can be executed early before the end date if the vote outcome cannot mathematically change by more voters voting.
+    /// @param VoteReplacment In vote replacement mode, voters can change their vote multiple times and only the latest vote option is tallied.
+    enum VotingMode {
+        Standard,
+        EarlyExecution,
+        VoteReplacement
+    }
+
+    /// @notice A container for the majority voting settings that will be applied as parameters on proposal creation.
+    /// @param votingMode A parameter to select the vote mode.
+    /// @param supportThreshold The support threshold value.
+    /// @param minParticipation The minimum participation value.
+    /// @param minDuration The minimum duration of the proposal vote in seconds.
+    /// @param minProposerVotingPower The minimum voting power required to create a proposal.
+    struct VotingSettings {
+        VotingMode votingMode;
+        uint64 supportThreshold;
+        uint64 minParticipation;
+        uint64 minDuration;
+        uint256 minProposerVotingPower;
+    }
+
+    /// @notice A container for proposal-related information.
+    /// @param executed Wheter the proposal is executed or not.
+    /// @param parameters The proposal parameters at the time of the proposal creation.
+    /// @param tally The vote tally of the proposal.
+    /// @param voters The votes casted by the voters.
+    /// @param actions The actions to be executed when the proposal passes.
+    struct Proposal {
+        bool executed;
+        ProposalParameters parameters;
+        Tally tally;
+        mapping(address => IMajorityVoting.VoteOption) voters;
+        IDAO.Action[] actions;
+    }
```

Comment:
Wouldn't this belong to the Proposal interface/base-contract?

Reply:
It cannot because the struct will differ for every governance plugin.

In my PR, I took the best compromise that was possible with our current architecture. I reverted it as an outcome of this discussion https://github.com/aragon/core/pull/167#discussion_r1054898602 to be done in a separate task.

Reply:
Thank you for the explanation Michael